addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/handler/YeelightHandlerBase.java

```diff
@@ -259,23 +259,19 @@ public void onStatusChanged(String prop, DeviceStatus status) {
     }
 
     void updateBrightnessAndColorUI(DeviceStatus status) {
-        if (status.isPowerOff()) {
-            updateState(CHANNEL_BRIGHTNESS, new PercentType(0));
-        } else {
-            updateState(CHANNEL_BRIGHTNESS, new PercentType(status.getBrightness()));
-            HSBType hsbType = null;
-            if (status.getMode() == DeviceMode.MODE_COLOR) {
-                hsbType = HSBType.fromRGB(status.getR(), status.getG(), status.getB());
-            } else if (status.getMode() == DeviceMode.MODE_HSV) {
-                hsbType = new HSBType(new DecimalType(status.getHue()), new PercentType(status.getSat()),
-                        new PercentType(1));
-            }
-            if (hsbType != null) {
-                updateState(CHANNEL_COLOR, hsbType);
-            }
-            updateState(CHANNEL_COLOR_TEMPERATURE,
-                    new PercentType((status.getCt() - COLOR_TEMPERATURE_MINIMUM) / COLOR_TEMPERATURE_STEP));
-        }
+        PercentType brightness = status.isPowerOff() ? new PercentType(0) : new PercentType(status.getBrightness());
+
+        HSBType tempHsbType = HSBType.fromRGB(status.getR(), status.getG(), status.getB());
+        HSBType hsbType = status.getMode() == DeviceMode.MODE_HSV
+                ? new HSBType(new DecimalType(status.getHue()), new PercentType(status.getSat()), brightness)
+                : new HSBType(tempHsbType.getHue(), tempHsbType.getSaturation(), brightness);
+
+        logger.debug("Update Color->{}", hsbType.toString());
```

Comment:
Don't call `.toString()` on parameter objects when logging. This will automatically be done by the logger (but only if the configured log level is high enough). This way the relatively expensive `toString` operations are only executed when strictly necessary.

Reply:
Sure.